<?xml version="1.0"?>
<launch>
    
  <param name="position" value="1"/>
  <param name="velocity" value="0"/>


  <rosparam file="$(find flippers_test)/config/controllers.yaml" command="load"/>
  <rosparam file="$(find flippers_test)/config/joint_limits.yaml" command="load"/>
   

   <!--
  <arg name="model" default="$(find flippers_test)/urdf/single_joint_actuator.urdf.xacro"/>
  <arg name="gui" default="true" />
  
  <param name="robot_description" command="$(find xacro)/xacro.py $(arg model)" />
  <param name="use_gui" value="$(arg gui)"/>
  -->

  <node name="flippers_node" pkg="flippers_test" type="flippers_node" output="screen">
      <param name="front_left" type="int" value="11" />
      <param name="front_left_drive_upper_limit" type="double" value="85000" />
      <param name="front_left_drive_lower_limit" type="double" value="15000" />
      <param name="front_left_drive_peak_output_forward" type="double" value="0.325" />
      <param name="front_left_drive_peak_output_reverse" type="double" value="-0.36" />

      <param name="front_right" type="int" value="12" />
      <param name="front_right_drive_upper_limit" type="double" value="85000" />
      <param name="front_right_drive_lower_limit" type="double" value="15000" />
      <param name="front_right_drive_peak_output_forward" type="double" value="0.325" />
      <param name="front_right_drive_peak_output_reverse" type="double" value="-0.36" />

      <param name="rear_left" type="int" value="13" />
      <param name="rear_left_drive_upper_limit" type="double" value="85000" />
      <param name="rear_left_drive_lower_limit" type="double" value="15000" />
      <param name="rear_left_drive_peak_output_forward" type="double" value="0.35" />
      <param name="rear_left_drive_peak_output_reverse" type="double" value="-0.38" />

      <param name="rear_right" type="int" value="14" />
      <param name="rear_right_drive_upper_limit" type="double" value="85000" />
      <param name="rear_right_drive_lower_limit" type="double" value="15000" />
      <param name="rear_right_drive_peak_output_forward" type="double" value="0.36" />
      <param name="rear_right_drive_peak_output_reverse" type="double" value="-0.30" />

      <param name="kP" type="double" value="1.27" />
      <param name="kI" type="double" value="0" />
      <param name="kD" type="double" value="0" />
  
</node>


  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" >
  </node>

<!--
  <node name="rviz" pkg="rviz" type="rviz"/>
-->
        
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
        args="
        	/markhor/joints_update
          /markhor/flipper_fl_position_controller
          /markhor/flipper_fr_position_controller
          /markhor/flipper_rl_position_controller
          /markhor/flipper_rr_position_controller"/>
<!--
  <node name="controller_spawner_stopped" pkg="controller_manager" type="spawner" respawn="false" output="screen"
        args="-stopped/single_joint_actuator/joint1_velocity_controller"/>
-->


</launch>
